File: backend/src/main/java/com/sichrplace/admin/AdminService.java
package com.sichrplace.admin;

import com.sichrplace.common.dto.ApiResponse;
import com.sichrplace.admin.dto.SupportTicketStatusRequest;
import java.util.List;

public interface AdminService {
    ApiResponse<List<String>> listUsers();
    ApiResponse<Void> updateUserStatus(Long id, SupportTicketStatusRequest request);
    ApiResponse<List<String>> listTickets();
    ApiResponse<Void> updateTicketStatus(Long id, SupportTicketStatusRequest request);
}

File: frontend/src/services/adminService.ts
import api from '../config/api';
import type { ApiResponse } from '../types/api';

export interface UserSummary {
  id: number;
  email: string;
  role: string;
}

export interface TicketSummary {
  id: number;
  subject: string;
  status: string;
}

export interface StatusUpdateRequest {
  status: string;
}

export async function listUsers(): Promise<ApiResponse<UserSummary[]>> {
  const { data } = await api.get<ApiResponse<UserSummary[]>>('/api/admin/users');
  return data;
}

export async function updateUserStatus(id: string, payload: StatusUpdateRequest): Promise<ApiResponse<void>> {
  const { data } = await api.put<ApiResponse<void>>(`/api/admin/users/${id}/status`, payload);
  return data;
}

export async function listTickets(): Promise<ApiResponse<TicketSummary[]>> {
  const { data } = await api.get<ApiResponse<TicketSummary[]>>('/api/admin/tickets');
  return data;
}

export async function updateTicketStatus(id: string, payload: StatusUpdateRequest): Promise<ApiResponse<void>> {
  const { data } = await api.put<ApiResponse<void>>(`/api/admin/tickets/${id}/status`, payload);
  return data;
}
